client/txn.go

```diff
@@ -666,7 +666,12 @@ func (txn *Txn) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.
 	if elideEndTxn && pErr == nil {
 		// Check that read only transactions do not violate their deadline.
 		if endTxnRequest.Deadline != nil {
-			if endTxnRequest.Deadline.Less(txn.Proto.Timestamp) {
+			t := txn.Proto.Timestamp
```

Comment:
I don't think it's valid for `txn.Proto.Timestamp` to be zero. Such a transaction would always pass the deadline check, but would never be able to commit because it is always in the past. Timestamp and OrigTimestamp are both initialized in roachpb.NewTransaction.